src/catalog/src/durable/impls/persist/state_update.rs

```diff
@@ -542,124 +541,81 @@ impl RustType<proto::StateUpdateKind> for StateUpdateKind {
-        SimpleSchema::<StateUpdateKindBinary, Vec<u8>>::decoder(cols, |val, ret| {
-            *ret =
-                StateUpdateKindBinary::decode(val).expect("should be valid StateUpdateKindBinary")
-        })
+impl StateUpdateKindRaw {
+    pub fn from_serde<S: serde::Serialize>(s: &S) -> Self {
+        let kind = serde_json::to_value(s).expect("WIP");
+        let mut row = Row::default();
+        let mut packer = row.packer();
+        JsonbPacker::new(&mut packer)
+            .pack_serde_json(kind)
+            .expect("WIP");
+        StateUpdateKindRaw(SourceData(Ok(row)))
     }
 
-    fn encoder<'a>(&self, cols: ColumnsMut<'a>) -> Result<Self::Encoder<'a>, String> {
-        SimpleSchema::<StateUpdateKindBinary, Vec<u8>>::push_encoder(cols, |col, val| {
-            let mut buf = Vec::new();
-            StateUpdateKindBinary::encode(val, &mut buf);
-            mz_persist_types::columnar::ColumnPush::<Vec<u8>>::push(col, &buf)
-        })
+    pub fn to_serde<D: serde::de::DeserializeOwned>(&self) -> D {
+        let row = self.0.as_ref().expect("WIP");
+        let kind = row.iter().exactly_one().expect("WIP");
+        let kind = JsonbRef::from_datum(kind).to_serde_json();
+        serde_json::from_value::<D>(kind).expect("WIP")
     }
 }
 
 #[cfg(test)]
 mod tests {
+    use crate::durable::impls::persist::state_update::StateUpdateKindRaw;
+    use crate::durable::impls::persist::StateUpdateKind;
     use mz_persist_types::Codec;
-    use mz_proto::ProtoType;
-    use mz_proto::RustType;
+    use mz_storage_types::sources::SourceData;
     use proptest::prelude::*;
-    use prost::Message;
-
-    use crate::durable::impls::persist::state_update::StateUpdateKindBinary;
-    use crate::durable::impls::persist::StateUpdateKind;
-    use crate::durable::objects::serialization::proto;
 
     proptest! {
         #[mz_ore::test]
         #[cfg_attr(miri, ignore)] // slow
         fn proptest_state_update_kind_roundtrip(kind: StateUpdateKind) {
-            // Roundtrip through `Vec<u8>`.
-            let encoded = kind.into_proto().encode_to_vec();
-            let decoded = proto::StateUpdateKind::decode(encoded.as_slice())
-                .expect("decode error")
-                .into_rust()
-                .expect("decode error");
-
-            prop_assert_eq!(&kind, &decoded);
+            let raw = StateUpdateKindRaw(kind.to_json_row());
 
-            // Roundtrip through `StateUpdateKindBinary`.
-            let kind_binary: StateUpdateKindBinary = kind.clone().into();
-            let kind_roundtrip = StateUpdateKind::try_from(kind_binary).expect("should roundtrip");
+            let mut encoded = Vec::new();
+            raw.0.encode(&mut encoded);
 
-            prop_assert_eq!(kind, kind_roundtrip);
-        }
-
-        #[mz_ore::test]
-        #[cfg_attr(miri, ignore)] // slow
-        fn proptest_state_update_kind_binary_roundtrip(kind: StateUpdateKindBinary) {
```

Comment:
I wasn't quite sure how to conceptually map these to the new stuff. Let me know if you think we lost any meaningful test coverage here